### ISSUE TITLE
Deprecation warning when building native addon since 5.x

### ISSUE THREAD

**Issue by logidelic** [electron/electron#18435]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:** 
  * 5.0.2
* **Operating System:**
  * Ubuntu 18.04 x64 (& arm64)
* **Last Known Working Electron version:**
  * 4.x

### Actual Behavior
When I build my native (C++) addon for Electron, using the latest node-gyp and the latest nan release, I get the following (and a few other) deprecation warnings:

```
gyp info it worked if it ends with ok
gyp info using node-gyp@4.0.0
gyp info using node@10.15.3 | linux | arm64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/user/myaddon/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/user/.node-gyp/5.0.2/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/user/.node-gyp/5.0.2',
gyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/user/.node-gyp/5.0.2/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/user/myaddon',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/user/myaddon/build'

  CXX(target) Release/obj.target/myaddon/addon.o
In file included from /home/user/.node-gyp/5.0.2/include/node/v8-internal.h:13:0,
                 from /home/user/.node-gyp/5.0.2/include/node/v8.h:25,
                 from /home/user/.node-gyp/5.0.2/include/node/node.h:63,
                 from ../node_modules/nan/nan.h:53,
                 from ../addon.cpp:20:
/home/user/.node-gyp/5.0.2/include/node/v8config.h:307:49: warning: ‘BufferReference’ is deprecated: Use MemorySpan<const uint8_t> [-Wdeprecated-declarations]
   declarator __attribute__((deprecated(message)))
                                                 ^
/home/user/.node-gyp/5.0.2/include/node/v8.h:4444:3: note: in expansion of macro ‘V8_DEPRECATED’
   V8_DEPRECATED("Use CompiledWasmModule::GetWireBytesRef()",
   ^
In file included from /home/user/.node-gyp/5.0.2/include/node/node.h:63:0,
                 from ../node_modules/nan/nan.h:53,
                 from ../addon.cpp:20:
/home/user/.node-gyp/5.0.2/include/node/v8.h:4386:58: note: declared here
   V8_DEPRECATED("Use MemorySpan<const uint8_t>", struct) BufferReference {
                                                          ^~~~~~~~~~~~~~~
In file included from /home/user/.node-gyp/5.0.2/include/node/v8-internal.h:13:0,
                 from /home/user/.node-gyp/5.0.2/include/node/v8.h:25,
                 from /home/user/.node-gyp/5.0.2/include/node/node.h:63,
                 from ../node_modules/nan/nan.h:53,
                 from ../addon.cpp:20:
```

### To Reproduce
Build a native addon using node-gyp for Electron 5 using a command line such as:
```
node-gyp rebuild --release --target=5.0.2 --arch=x64 --dist-url=https://electronjs.org/headers
```
This is just as documented here:
https://github.com/electron/electron/blob/master/docs/tutorial/using-native-node-modules.md

### Additional Information
I see in the `node-gyp` output that it seems to be building against node `10.15.3` even though electron 5.0.2 is actually at node at node `12`. Is that the issue? If so, what's wrong with my build line?

I've been building native addons for ~2 years and never ran into this issue until I switched to Electron 5...

Thanks!

**Comment by MarshallOfSound** [comment]
@logidelic These warnings are to be expected, V8 has deprecated some APIs, you need to stop using the APIs that V8 has flagged as deprecated.  If you try build your module against node 12 you will probably see the exact same warnings.

**Comment by logidelic** [comment]
Hey @MarshallOfSound , I appreciate the response. Unfortunately it's not so simple. I'm not (directly?) using any of the referenced symbols and I'm not including any node_modules other than nan. The frustrating thing is that the warnings don't really give a hint as to where these deprecated calls are supposedly being used. :(

For what it's worth, I have plenty of experience

**Comment by MarshallOfSound** [comment]
@logidelic What native module is this? sometimes it's hard to follow the deprecation traces 😄

**Comment by logidelic** [comment]
It's my own native module. Obviously if no one else is getting this it must be my fault. I'll keep looking. Thanks. :)

**Comment by MarshallOfSound** [comment]
@logidelic I'd start with whatever is on line 20 of `addon.cpp` 😄

**Comment by logidelic** [comment]
Hah, it's the `nan.h` include... Hence my confusion about what the error is all about. :)

**Comment by nornagon** [comment]
This looks like a nan issue to me; try updating your version of nan and if that doesn't fix it then file a bug against nan :)
